File: core/src/main/java/com/emarsys/core/util/log/entry/OnScreenTime.java
package com.emarsys.core.util.log.entry;

import java.util.HashMap;
import java.util.Map;

public class OnScreenTime implements LogEntry {

    private final Map<String, Object> data;

    public OnScreenTime(long onScreenTime, long startScreenTime, long endScreenTime, String campaignId, String requestId) {
        data = new HashMap<>();
        data.put("campaign_id", campaignId);
        data.put("duration", onScreenTime);
        data.put("start", startScreenTime);
        data.put("end", endScreenTime);
        if (requestId == null) {
            data.put("source", "push");
        } else {
            data.put("source", "customEvent");
            data.put("request_id", requestId);
        }
    }

    @Override
    public String getTopic() {
        return "log_inapp_on_screen_time";
    }

    @Override
    public Map<String, Object> getData() {
        return data;
    }
}

File: core/src/main/java/com/emarsys/core/util/log/entry/OnScreenTime.kt
package com.emarsys.core.util.log.entry


class OnScreenTime(onScreenTime: Long, startScreenTime: Long, endScreenTime: Long, campaignId: String, requestId: String?) : LogEntry {
    override val data: Map<String, Any>
    override val topic: String
        get() = "log_inapp_on_screen_time"

    init {
        data = mutableMapOf(
                "campaign_id" to campaignId,
                "duration" to onScreenTime,
                "start" to startScreenTime,
                "end" to endScreenTime
        )
        if (requestId == null) {
            data["source"] = "push"
        } else {
            data["source"] = "customEvent"
            data["request_id"] = requestId
        }
    }
}